### PR TITLE
Feature: update linting

### PR DESCRIPTION
**Pull request by jamesjwarren**
This PR removes CSSComb as we felt it was hindering rather than helping our workflow, `lesslint` should suffice for css linting. I have also swapped out `jsbeautifier` for `htmlhint` as it provides more useful hints without the need to write to our source files.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -342,30 +342,16 @@ module.exports = function (grunt) {
...
-        }
-      }
+    /**
+     * HTML linter
+     * {@link https://github.com/htmllint/htmllint/wiki/Options .htmllintrc options}
+     */
+    htmllint: {
+      options: {
+        force: true,
+        htmllintrc: 'app/partials/.htmllintrc'
```
Comment (edwardoparearyee) [inline Gruntfile.js]: could you use `./` before directories/file paths

File: Gruntfile.js [hunk 2]
```diff
@@ -386,13 +372,11 @@ module.exports = function (grunt) {
...
   grunt.loadNpmTasks('grunt-lesslint');
-  grunt.loadNpmTasks('grunt-csscomb');
-  grunt.loadNpmTasks('grunt-jsbeautifier');
+  grunt.loadNpmTasks('grunt-htmllint');
 
   grunt.registerTask('build', [
     'clean:beforeBuild',
-    'jsbeautifier:html',
-    'csscomb',
+    'htmllint',
```
Comment (edwardoparearyee) [inline Gruntfile.js]: don't think we need to do any linting in the build task

File: Gruntfile.js [hunk 3]
```diff
@@ -412,8 +396,7 @@ module.exports = function (grunt) {
   ]);
 
   grunt.registerTask('server', [
-    'jsbeautifier:html',
-    'csscomb',
+    'htmllint',
```
Comment (edwardoparearyee) [inline Gruntfile.js]: Here linting is done in the `watch` task so don't really need to do it before calling `watch`. Also if the linting fails here the `watch` task won't run which means running the task manually everytime until it's fixed.

File: Gruntfile.js [hunk 4]
```diff
@@ -432,8 +414,7 @@ module.exports = function (grunt) {
   ]);
 
   grunt.registerTask('serverall', [
-    'jsbeautifier:html',
-    'csscomb',
+    'htmllint',
```
Comment (edwardoparearyee) [inline Gruntfile.js]: same goes for the other server tasks

File: Gruntfile.js [hunk 5]
```diff
@@ -443,7 +424,7 @@ module.exports = function (grunt) {
 
   grunt.registerTask('test', [
     'clean:beforeBuild',
-    'jsbeautifier:html',
+    'htmllint',
     'lesslint',
```
Comment (edwardoparearyee) [inline Gruntfile.js]: The `test` task is mainly to test js and are run with the `watch:js` task so maybe here we need another task for linting html and css files.

**Comment by jamesjwarren** [comment]
@edwardoparearyee I've split out the linting tasks into there own `lint` task which also runs in the `watch:css`

**Comment by edwardoparearyee** [comment]
LGTM :+1:

**Pull request merged**